x-pack/elastic-agent/pkg/agent/cmd/install.go

```diff
@@ -114,13 +118,13 @@ func installCmd(streams *cli.IOStreams, cmd *cobra.Command, args []string) error
 			enroll = false
 		}
 	}
-	if !askEnroll && (url == "" || token == "") {
+	if !askEnroll && (url == "" || token == "") && fleetServer == "" {
 		// force was performed without required enrollment arguments, all done (standalone mode)
 		enroll = false
 	}
 
 	if enroll {
-		if url == "" {
+		if url == "" && fleetServer == "" {
```

Comment:
could we just move this to `enroll && fleetServer == ""`  and remove also from the if above
or if the goal here is to skip enroll for fleet server maybe we can just extract this up to the line 140 into a Fn and execute only when `fleetServer == ""` 
these ifs are getting complicated

Reply:
I was able to simplify this by moving it up. The goal here is to ask for URL and Token in the case that it is not provided. The complexity comes from `install` being interactive.